File: macos/Scanner/Sources/Scanner/Rules/AESAUTH003.swift
public struct AESAUTH003: Rule {
    public let id = "AES-AUTH-003"
    public let severity: Severity = .medium

    public init() {}

    public func evaluate(_ facts: ScanFacts) -> [Finding] {
        facts.authFiles.map { authFile in
            finding(app: authFile.appId, affectedPath: authFile.filePath)
        }
    }
}

File: windows/Scanner/Rules/AESAUTH003.cs
using AIExposureScanner.Scanner.Models;

namespace AIExposureScanner.Scanner.Rules;

public sealed class AESAUTH003 : IRule
{
    public string Id => "AES-AUTH-003";
    public Severity Severity => Severity.Medium;

    public IReadOnlyList<Finding> Evaluate(ScanFacts facts) =>
        facts.AuthFiles
            .Select(authFile => this.Finding(authFile.AppId, affectedPath: authFile.FilePath))
            .ToArray();
}
